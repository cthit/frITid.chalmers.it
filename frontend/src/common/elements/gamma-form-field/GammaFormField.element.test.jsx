import React from "react";
import { shallow } from "enzyme";
import GammaFormField from "./";
import GammaCheckbox from "../gamma-checkbox";
import GammaTextField from "../gamma-text-field";

describe("<GammaFormField/>", () => {
  test("Shallow render of <GammaFormField/> with a GammaTextField", () => {
    const wrapper = shallow(
      <GammaFormField
        name="password"
        component={GammaTextField}
        componentProps={{ upperLabel: "this is password yes" }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of <GammaFormField/> with a GammaCheckbox", () => {
    const wrapper = shallow(
      <GammaFormField
        name="userAgreement"
        component={GammaCheckbox}
        componentProps={{
          primary: true,
          label: "Jag accepterar anvÃ¤ndaravtalen"
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
