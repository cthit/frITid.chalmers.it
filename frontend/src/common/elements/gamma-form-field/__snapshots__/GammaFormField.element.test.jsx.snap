// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GammaFormField/> Shallow render of <GammaFormField/> with a GammaCheckbox 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaFormField
    component={[Function]}
    componentProps={
        Object {
            "label": "Jag accepterar anvÃ¤ndaravtalen",
            "primary": true,
          }
    }
    name="userAgreement"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "name": "userAgreement",
      "render": [Function],
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "name": "userAgreement",
        "render": [Function],
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaFormField/> Shallow render of <GammaFormField/> with a GammaTextField 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaFormField
    component={[Function]}
    componentProps={
        Object {
            "upperLabel": "this is password yes",
          }
    }
    name="password"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "name": "password",
      "render": [Function],
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "name": "password",
        "render": [Function],
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
